tests/test_aea.py

```diff
@@ -180,3 +187,110 @@ def test_handle():
     finally:
         agent.stop()
         t.join()
+
+
+class TestInitializeAEAProgrammaticallyFromResourcesDir:
+    """Test that we can initialize the agent by providing the resource object loaded from dir."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set the test up."""
+        cls.agent_name = "MyAgent"
+        cls.private_key_pem_path = os.path.join(CUR_PATH, "data", "priv.pem")
+        cls.wallet = Wallet({'default': cls.private_key_pem_path})
+        cls.ledger_apis = LedgerApis({})
+        cls.connection = DummyConnection()
+        cls.mailbox = MailBox(cls.connection)
+
+        cls.resources = Resources(os.path.join(CUR_PATH, "data", "dummy_aea"))
+        cls.aea = AEA(cls.agent_name, cls.mailbox, cls.wallet, cls.ledger_apis, cls.resources)
+        cls.resources.load(cls.aea.context)
+
+        cls.expected_message = DefaultMessage(type=DefaultMessage.Type.BYTES, content=b"hello")
+        cls.connection.receive(Envelope(to="", sender="", protocol_id="default", message=DefaultSerializer().encode(cls.expected_message)))
+
+        cls.t = Thread(target=cls.aea.start)
+        cls.t.start()
+
+    def test_initialize_aea_programmatically(self):
+        """Test that we can initialize an AEA programmatically."""
+        time.sleep(1.0)
+
+        dummy_behaviour = next(filter(lambda x: x.__class__.__name__ == "DummyBehaviour", self.aea.resources.behaviour_registry.fetch("dummy")), None)
+        assert dummy_behaviour is not None
+        assert dummy_behaviour.nb_act_called > 0
+
+        dummy_task = next(filter(lambda x: x.__class__.__name__ == "DummyTask", self.aea.resources.task_registry.fetch("dummy")), None)
+        assert dummy_task is not None
+        assert dummy_task.nb_execute_called > 0
+
+        dummy_handler = next(filter(lambda x: x.__class__.__name__ == "DummyHandler", self.aea.resources.handler_registry.fetch("default")), None)
+        assert dummy_handler is not None
+        assert len(dummy_handler.handled_messages) == 1
+        assert dummy_handler.handled_messages[0] == self.expected_message
+
+    @classmethod
+    def teardown_class(cls):
+        """Tear the test down."""
+        cls.aea.stop()
+        cls.t.join()
+
+
+class TestInitializeAEAProgrammaticallyBuildResources:
+    """Test that we can initialize the agent by building the resource object."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set the test up."""
+        cls.agent_name = "MyAgent"
+        cls.private_key_pem_path = os.path.join(CUR_PATH, "data", "priv.pem")
+        cls.wallet = Wallet({'default': cls.private_key_pem_path})
+        cls.ledger_apis = LedgerApis({})
+        cls.connection = DummyConnection()
+        cls.mailbox = MailBox(cls.connection)
+
+        cls.temp = tempfile.mkdtemp(prefix="test_aea_resources")
+        cls.resources = Resources(cls.temp)
+        cls.aea = AEA(cls.agent_name, cls.mailbox, cls.wallet, cls.ledger_apis, resources=cls.resources)
```

Comment:
@dishmop this is now the new way to build an AEA programmatically. Any feedback?